typescript/src/ethereum.ts

```diff
@@ -488,6 +488,39 @@ export class Bridge
     redeemerOutputScript: string,
     amount: BigNumber
   ): Promise<void> {
+    const redemptionData = this.parseRequestRedemptionTransactionData(
+      walletPublicKey,
+      mainUtxo,
+      redeemerOutputScript
+    )
+
+    await sendWithRetry<ContractTransaction>(async () => {
+      return await this._instance.requestRedemption(
+        redemptionData.walletPublicKeyHash,
+        redemptionData.mainUtxo,
+        redemptionData.prefixedRawRedeemerOutputScript,
+        amount
+      )
+    }, this._totalRetryAttempts)
+  }
+
+  /**
+   * Parses the request redemption data to the proper form.
+   * @param walletPublicKey - The Bitcoin public key of the wallet. Must be in
+   *        the compressed form (33 bytes long with 02 or 03 prefix).
+   * @param mainUtxo - The main UTXO of the wallet. Must match the main UTXO
+   *        held by the on-chain contract.
+   * @param redeemerOutputScript - The output script that the redeemed funds
+   *        will be locked to. Must be un-prefixed and not prepended with
+   *        length.
+   * @returns Parsed data that can be passed to the contract to request
+   * redemption.
+   */
+  private parseRequestRedemptionTransactionData = (
```

Comment:
I am not sure *parsing* verb is correct in this context. I would expect parsing to happen when we receive on-chain data and we want them to be translated into domain-level objects.

If we make the change from the previous comment and keep `buildRedemptionData` private, then I would suggest to rename this function to 
`buildBridgeRequestRedemptionData` and `buildRedemptionData` to `buildTbtcTokenRequestRedemptionData`.

Reply:
9249ee5. I renamed the `buildRedemptionData` -> `buildRequestRedemptionData` because it lives inside the `TBTCToken` handle so IMO there is no need to add `TbtcToken`. WDYT?

Reply:
Yeah, it's in a separate class, this makes sense!